### DISCUSSION TITLE
``pyrx_onload.py``

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#147] [Ideas]
_Based on issue #145_ 

> loader behavior, will search local ..\ site-packages\pyrx for pyrx_onload fist, otherwise fall back to CAD’s search path settings

It is not a good idea for the user to place anything in the package directory (`` ...\site-packages\pyrx``). Therefore, it should be assumed that ``pyrx_onload.py`` present in this directory will be added by the `pyrx` developers. User files should be given priority, so CAD search paths should be searched first, and the `pyrx` directory last.

Additionally, you should consider whether if the `site-packages\pyrx\pyrx_onload.py` file exists, should the user file overwrite it. If such a file were to exist, it would probably do something important that should not be omitted. Any omission of loading the original `pyrx_onload.py` file should be explicitly defined (via an environment variable or something similar).

**Comment by CEXT-Dan** [comment]
I guess I could have worded it better, it’s the default behavior of HostApplicationServices.FindFile. 
You’re right it should not be installed in ...\site-packages\pyrx, nor should the user add it

There’s an internal pyrx_onload.py that gets copied to the debug development folder for running unit tests
